src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/OptionDefinition.cs

```diff
@@ -71,8 +65,7 @@ public bool Equals(OptionDefinition other)
         {
             var equals = this.Name == other.Name &&
                 this.Feature == other.Feature &&
-                this.Group == other.Group &&
-                this.IsPerLanguage == other.IsPerLanguage;
```

Comment:
Theoretically, this is a breaking change as if we had two options, say one an `Option2<T>` and one a `PerLanguageOption2<T>`, such that they had the same `feature`, `group`, `name`, `defaultValue` and type `T`, their underlying `OptionDefinition` was non-equal due to `isPerLanguage` flag, but now it would be equal. I don't think we have or want to have any such same named options though, so this should be fine.